pkg/node/manager/manager.go

```diff
@@ -572,14 +572,17 @@ func (m *Manager) StartNeighborRefresh(nh datapath.NodeHandler) {
 				m.mutex.RLock()
 				defer m.mutex.RUnlock()
 				for _, entry := range m.nodes {
-					if entry.node.IsLocal() {
+					entry.mutex.Lock()
+					entryNode := entry.node
+					entry.mutex.Unlock()
+					if entryNode.IsLocal() {
```

Comment:
`IsLocal()` is a method of `*nodeType.Node`, while `entryNode` is `nodeType.Node`. I'm puzzled why it compiles.

Reply:
You can access methods of a type regardless the receiver being a pointer or not.